crates/nao_camera/src/parameters.rs

```diff
@@ -0,0 +1,54 @@
+use serde::Deserialize;
+
+use crate::bindings::{
+    v4l2_exposure_auto_type_V4L2_EXPOSURE_APERTURE_PRIORITY,
+    v4l2_exposure_auto_type_V4L2_EXPOSURE_AUTO, v4l2_exposure_auto_type_V4L2_EXPOSURE_MANUAL,
+    v4l2_exposure_auto_type_V4L2_EXPOSURE_SHUTTER_PRIORITY,
+};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Parameters {
+    pub width: u32,
+    pub height: u32,
+    pub format: Format,
+    pub time_per_frame: Fraction,
+
+    pub brightness: i32,
+    pub contrast: i32,
+    pub saturation: i32,
+    pub hue: i32,
+    pub white_balance_temperature_auto: bool,
+    pub gain: i32,
+    pub hue_auto: bool,
+    pub white_balance_temperature: i32,
+    pub sharpness: i32,
+    pub exposure_auto: ExposureMode,
+    pub exposure_absolute: i32,
+    pub focus_absolute: i32,
+    pub focus_auto: bool,
+
+    pub automatic_exposure_control_weights: [u8; 16],
+    pub disable_digital_effects: bool,
+    pub flip_sensor: bool,
+
+    pub amount_of_buffers: u32,
```

Comment:
Not planned currently because there is no change mechanism implemented (would require e.g. to restart the camera with new parameters etc.). In addition, this would require special handling in Communication and would therefore break robotics/framework separation. In addition, changing parameters without restarting the framework is not a common use case.